KmsClient.js

```diff
@@ -93,9 +92,17 @@ export class KmsClient {
       if(e.status === 409) {
         const err = new Error('Duplicate error.');
         err.name = 'DuplicateError';
-        throw err;
+        err.cause = e;
+        _handleClientError({
+          message: 'Duplicate error while generating key.',
+          cause: e
+        });
       }
-      throw e;
+
+      _handleClientError({
+        message: 'Error generating key.',
+        cause: e
+      });
```

Comment:
I recommend simplifying this whole `catch` block to:

```js
let cause;
if(e.status === 409) {
  cause = new Error('Duplicate error.');
  cause.name = 'DuplicateError';
  cause.cause = e;
} else {
  cause = e;
}

_handleClientError({
  message: 'Error generating key.',
  cause
});
```

Reply:
Changed, thanks.